iree/turbine/kernel/wave/cache.py

```diff
@@ -88,10 +88,28 @@ def extract_free_vars(kernel_fn: Callable):
     return [
         (k, v)
         for k, v in inspect.getclosurevars(kernel_fn).nonlocals.items()
-        if not isinstance(v, IndexMapping)
+        if not isinstance(v, (IndexMapping, Callable))
     ]
 
 
+def get_nested_functions(root_fn: Callable):
+    """Simple BFS search to get all sub functions inside a wave kernel."""
+    workqueue = deque([root_fn])
+    fn_list = set([root_fn])
+    while len(workqueue) > 0:
```

Comment:
nit: `while workqueue:`

Reply:
ooo nice! didn't know we could do that

Reply:
done!

Reply:
yeah, all containers treated as `False` when empty and `True` otherwise, when casted to bool.